### PR TITLE
Reactification

### PR DESCRIPTION
**Pull request by peric**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
+    <meta name="apple-mobile-web-app-capable" content="yes">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta name="description" content="Get countries list from geonames">
+    <meta name="author" content="Drazen Peric">
+    <meta name="keywords" content="countries, list, mysql, firebird, geonames, xml, csv, yaml">
+    <title>getCountries()</title>
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap.min.css">
+    <link rel="stylesheet" type="text/css" href="css/style.css" />
+    <script src="https://fb.me/react-0.13.2.js"></script>
+    <script src="https://fb.me/JSXTransformer-0.13.2.js"></script>
```
Comment (zigomir) [inline index.html]: you wan't to have build process done locally and not running transformer on other people's machines ;)

File: index.html [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
+    <script src="https://code.jquery.com/jquery-2.1.3.min.js"></script>
```
Comment (zigomir) [inline index.html]: Not sure, but do you need jquery?

File: js/app.js [hunk 3]
```diff
@@ -0,0 +1,548 @@
+var OUTPUT_MYSQL = 'MySQL',
```
Comment (zigomir) [inline js/app.js]: This file can easily be split to few smaller files.

File: js/app.js [hunk 4]
```diff
@@ -0,0 +1,548 @@
...
+
+            data[objectKey].checked = !data[objectKey].checked;
+        } else {
+            data[objectKey].checked = !data[objectKey].checked;
+        }
+
+        this.forceUpdate();
+
+        // TODO: do I need this? How forceUpdate does updates based on references?
+//        this.setState({type: data});
```
Comment (zigomir) [inline js/app.js]: Use `setState` wherever you can. `forceUpdate` should be your last exit. `setState` will trigger re-render of component and this is usually what you want.

File: js/app.js [hunk 5]
```diff
@@ -0,0 +1,548 @@
...
+            return (
+                <Setting
+                    key={index + key}
+                    objectKey={key}
+                    data={self.state.settings}
+                    name={setting.name}
+                    longName={setting.longName}
+                    checked={setting.checked}
+                    onChange={self.toggleCheck}
+                    disabled={setting.disabled} />
```
Comment (zigomir) [inline js/app.js]: props for using lots of `props` :+1: :)

File: js/app.js [hunk 6]
```diff
@@ -0,0 +1,548 @@
...
+    },
+    render: function() {
+        var self = this;
+        var columns = Object.keys(this.state.columns).map(function(key, index) {
+            var column = self.state.columns[key];
+            return (
+                <Column
+                    key={index + column.name}
+                    objectKey={key}
+                    data={self.state.columns}
```
Comment (zigomir) [inline js/app.js]: `data` is really generic name. I believe you can think of something more specific ;)

File: js/app.js [hunk 7]
```diff
@@ -0,0 +1,548 @@
...
+                        data-type="settings"
+                        checked={this.props.checked}
+                        onChange={this.handleChange}
+                        disabled={this.props.disabled} />
+                            {this.props.longName}
+                </label>
+            </div>
+        );
+    }
+});
```
Comment (zigomir) [inline js/app.js]: Liking small components like `Setting` and `Column` you did here :+1:

File: js/app.js [hunk 8]
```diff
@@ -0,0 +1,548 @@
...
+                    "{0}" +
+                    "    PRIMARY KEY (`id`)\n" +
+                    ") ENGINE=MyISAM DEFAULT CHARSET=utf8 AUTO_INCREMENT=0;\n\n" +
+                    "" +
+                    "CREATE TABLE IF NOT EXISTS `country_lang_lnk` (\n" +
+                    "{1}" +
+                    "    PRIMARY KEY (`countryCode`,`languages`)\n" +
+                    ") ENGINE=MyISAM DEFAULT CHARSET=utf8;\n\n" +
+                    "{2}\n" +
+                    "{3}";
```
Comment (zigomir) [inline js/app.js]: ohnoes, this hurts. We're gonna convert this project to ES6 so you can use [template strings](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/template_strings)

File: js/main.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+(function() {
+    var outputCode = document.getElementById('outputcode');
+
+    // select on focus
+    outputCode.onfocus = function() {
+        this.select();
+
+        // TODO: copy to clipboard
+        // https://github.com/zeroclipboard/zeroclipboard
```
Comment (zigomir) [inline js/main.js]: :+1:

**Comment by peric** [comment]
@zigomir, thanks a lot for that useful review and PR! :)

**Comment by zigomir** [comment]
hhh :+1:

**Pull request merged**